lib/WTSI/NPG/Data/ConsentWithdrawn.pm

```diff
@@ -0,0 +1,310 @@
+has '_files' => (
+  isa        => 'ArrayRef',
+  is         => 'ro',
+  required   => 0,
+  lazy_build => 1,
+);
+sub _build__files {
+  my $self = shift;
+
+  my $iquest_cmd = $self->_iquery;
+  $self->info($iquest_cmd);
+  my @files;
+  my $no_rows_error = 0;
+
+  my $pid = open3( undef, my $iquest_out_fh, undef, $iquest_cmd);
+  while (my $line = <$iquest_out_fh> ) {
+    chomp $line;
+    if($line =~ /^ERROR/mxs) {
+      if ($line !~ /CAT_NO_ROWS_FOUND/mxs) {
+        $self->debug($line);
+      } else {
+        $no_rows_error = 1;
+      }
+      last;
+    }
+    if ($line =~/[.]bam$|[.]cram$/mxs) {
+      $self->debug('Found ' . $line);
+      push @files, $line;
+    }
+  }
+
+  waitpid $pid, 0;
+  if( $CHILD_ERROR >> $EXIT_CODE_SHIFT ) {
+    if (!$no_rows_error) {
+      $self->logcroak("Failed: $iquest_cmd");
+    }
+  }
+  close $iquest_out_fh or
+    $self->logcroak("Cannot close iquest command output: $ERRNO");
+
+  @files = sort @files;
+  return \@files;
+}
+
+# List of files with sample_consent_withdrawn flag set
+# and sample_consent_withdrawn_email_sent not set
+has '_new_files' => (
+  isa        => 'ArrayRef',
+  is         => 'ro',
+  required   => 0,
+  lazy_build => 1,
+);
+sub _build__new_files {
+  my $self = shift;
+
+  my @files = ();
+  foreach my $file (@{$self->_files}) {
+    my @meta =
+      grep { $_->{value} }
+      grep { $_->{attribute} eq $RT_TICKET_FLAG_META_KEY }
+      $self->irods->get_object_meta($file);
+    @meta or push @files, $file;
+  }
+
+  return \@files;
+}
+
+sub _create_rt_ticket {
+  my $self = shift;
+
+  my $payload  = "The files with consent withdrawn:\n\n" . join qq[\n] , @{$self->_new_files};
+  $self->info(qq{The following email will be sent to $RT_TICKET_EMAIL_ADDRESS\n$payload\n});
+
+  $self->dry_run and return;
+
+  MIME::Lite->new(
+    To            => $RT_TICKET_EMAIL_ADDRESS,
+    Subject       => q{iRODS files: sample consent withdrawn},
+    Type          => 'TEXT',
+    Data          => $payload,
+  )->send();
+
+  return;
+}
+
+sub _iquery {
+  my $self = shift;
+
+  my $query = q{iquest --no-page};
+  if ($self->zone) {
+    $query .= q[ -z ] .$self->zone;
+  }
+  $query .=
+    q{ "%s/%s" "select COLL_NAME, DATA_NAME where META_DATA_ATTR_NAME = 'sample_consent_withdrawn'} .
+    q{ and META_DATA_ATTR_VALUE = '1' and DATA_NAME not like '%header.bam%'};
+  if ($self->collection) {
+    $query .= q{ and COLL_NAME like '} . $self->collection() . q{%'};
+  }
+  $query .= q{"};
+
+  return $query;
+}
+
+sub _restrict_permissions {
+  my ($self, $file) = @_;
+
+  my @files = ( $file );
+
+  my $index_file = $file;
+  $index_file =~ s/[.]bam$/.bai/smx;
+  $index_file =~ s/[.]cram$/.cram.crai/smx;
+  my $obj = WTSI::NPG::HTS::DataObject->new($self->irods, $index_file);
+  if ($obj->is_present) {
+    push @files, $index_file;
+  }
+
+  my $nullp = $WTSI::NPG::iRODS::NULL_PERMISSION;
+  for my $f (@files) {
+    my @to_remove = grep { $_->{level} ne 'own' }
```

Comment:
`$WTSI::NPG::iRODS::OWN_PERMISSION` vs. `own`?

Reply:
Resolved in 6ca34f2